rs/consensus/src/idkg.rs

```diff
@@ -651,105 +630,113 @@ mod tests {
-        assert_eq!(args.requested_signatures.len(), 1);
-        assert_eq!(
-            args.requested_signatures.first().unwrap(),
-            &expected_request_id
-        );
+    }
+
+    fn get_fake_artifact_id_data(i: IDkgTranscriptId) -> IDkgArtifactIdData {
+        IDkgArtifactIdData {
+            height: i.source_height(),
+            subnet_id: *i.source_subnet(),
+            hash: CryptoHash(vec![]),
+        }
+    }
+
+    fn get_fake_share_id_data(i: &RequestId) -> SigShareIdData {
+        SigShareIdData {
+            hash: CryptoHash(vec![]),
+            height: i.height,
+        }
     }
 
     // Tests the priority computation for dealings/support.
     #[test]
     fn test_idkg_priority_fn_dealing_support() {
-        let xnet_subnet_id = SubnetId::from(PrincipalId::new_subnet_test_id(1));
-        let subnet_id = SubnetId::from(PrincipalId::new_subnet_test_id(2));
-        let xnet_transcript_id = IDkgTranscriptId::new(xnet_subnet_id, 1, Height::from(1000));
-        let transcript_id_fetch_1 = IDkgTranscriptId::new(subnet_id, 1, Height::from(80));
-        let transcript_id_drop = IDkgTranscriptId::new(subnet_id, 2, Height::from(70));
-        let transcript_id_fetch_2 = IDkgTranscriptId::new(subnet_id, 3, Height::from(102));
-        let transcript_id_stash = IDkgTranscriptId::new(subnet_id, 4, Height::from(200));
+        let xnet_transcript_id = IDkgTranscriptId::new(SUBNET_1, 1, Height::from(1000));
+        let local_subnet_id = SUBNET_2;
+        let transcript_id_fetch_1 = IDkgTranscriptId::new(local_subnet_id, 1, Height::from(80));
+        let transcript_id_fetch_2 = IDkgTranscriptId::new(local_subnet_id, 3, Height::from(102));
+        let transcript_id_stash = IDkgTranscriptId::new(local_subnet_id, 4, Height::from(200));
 
-        let metrics_registry = MetricsRegistry::new();
-        let metrics = IDkgGossipMetrics::new(metrics_registry);
-
-        let mut requested_transcripts = BTreeSet::new();
-        requested_transcripts.insert(transcript_id_fetch_1);
         let args = IDkgPriorityFnArgs {
             finalized_height: Height::from(100),
             certified_height: Height::from(100),
-            requested_transcripts,
-            requested_signatures: BTreeSet::new(),
-            active_transcripts: BTreeSet::new(),
         };
 
         let tests = vec![
             // Signed dealings
             (
-                IDkgMessageAttribute::Dealing(xnet_transcript_id),
+                IDkgMessageId::Dealing(
+                    dealing_prefix(&xnet_transcript_id, &NODE_1),
+                    get_fake_artifact_id_data(xnet_transcript_id).into(),
+                ),
                 Priority::FetchNow,
             ),
             (
-                IDkgMessageAttribute::Dealing(transcript_id_fetch_1),
+                IDkgMessageId::Dealing(
+                    dealing_prefix(&transcript_id_fetch_1, &NODE_1),
+                    get_fake_artifact_id_data(xnet_transcript_id).into(),
```

Comment:
Should it be the same transcript ID?
```suggestion
                    get_fake_artifact_id_data(transcript_id_fetch_1).into(),
```